doc/faq/installing_faq.rst

```diff
@@ -340,14 +205,71 @@ or::
 
 If you get the result ``/usr/bin/python2.7``, then you are getting the Python
 installed with OSX, which is probably not what you want.  Try closing and
-restarting Terminal.app before running the check again. If that doesn't fix
-the problem, depending on which Python you wanted to use, consider
-reinstalling Python.org Python, or check your homebrew or macports setup.
-Remember that the disk image installer only works for Python.org Python, and
-will not get picked up by other Pythons.  If all these fail, please let us
-know: see :ref:`reporting-problems`.
+restarting Terminal.app before running the check again. If that doesn't fix the
+problem, depending on which Python you wanted to use, consider reinstalling
+Python.org Python, or check your homebrew or macports setup.  Remember that
+the disk image installer only works for Python.org Python, and will not get
+picked up by other Pythons.  If all these fail, please :ref:`let us know
+<reporting-problems>`.
 
 Windows Notes
 =============
 
 See :ref:`installing_windows`.
+
+.. _install-from-git:
+
+Install from source
+===================
+
+Clone the main source using one of::
+
+   git clone git@github.com:matplotlib/matplotlib.git
+
+or::
+
+   git clone git://github.com/matplotlib/matplotlib.git
+
+and build and install as usual with::
+
+   cd matplotlib
+   pip install .
+
+.. note::
+
+   If you are on Debian/Ubuntu, you can get all the dependencies required to
+   build Matplotlib with::
+
+      sudo apt-get build-dep python-matplotlib
+
+   If you are on Fedora/RedHat, you can get all the dependencies required to
+   build matplotlib by first installing ``yum-builddep`` and then running::
+
+      su -c 'yum-builddep python-matplotlib'
+
+   This does not build Matplotlib, but it does get all of the build
+   dependencies, which will make building from source easier.
+
+If you want to be able to follow the development branch as it changes
+just replace the last step with::
+
+   pip install -e .
```

Comment:
One gotcha that I've run into is `pip` not being installed in the conda environment I'm in, so it uses the default `pip`, and installs the development version over the released version in my main conda environment.  Not sure where environments fit into this document, but something that could be discussed.

Reply:
I updated the thing to use `python -mpip` throughout, which should make sure pip always works from (and thus, in) the local environment.
I also think that conda now installs pip by default in all envs?  (but it's even if it doesn't it's not a problem as long as you use `python -mpip`).